### PR TITLE
MultiStore - Sell > Customer Service > Merchandise Return - Add Checkbox & Simplify Form

### LINKED ISSUE #23610: BO - Multistore - Merchandise Returns - The Multistore block & checkboxes are not displayed when selecting a shop
#### Describe the bug

Multistore enabled, In the BO > admin-dev/index.php/sell/customer-service/merchandise-return/ page when we select a shop, The multistore block and checkboxes are not displayed

#### Expected behavior

image

#### Steps to Reproduce

1. Enable Multistore Context
2. Create a new shop
3. Go to BO > admin-dev/index.php/sell/customer-service/merchandise-return/
4. Select a shop
5. See error: The multistore block and checkboxes are not displayed

Related to this PR: #15022
**Screenshots**

image

#### Additional information

* PrestaShop version: PS1772, develop
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | MultiStore - Sell > Customer Service > Merchandise Return - Add Checkbox & Simplify Form
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23610
| How to test?      | Cf #23610

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27720)

### CODE REVIEW

**Comment by ghost** [comment]
Since I saw @Kaudaj 's test, I do what he did 

https://github.com/PrestaShop/PrestaShop/pull/27651/files#diff-c5c0fbf09b409ff5e6293c86610cfc1106b847f8b501ca682700629208d56be9R52-R58

LGTM

**Review by ghost — DISMISSED** [review]

**Comment by Progi1984** [comment]
@okom3pom Fixed

**Review by ghost — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]
nice !

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php [hunk 1]
```diff
@@ -26,36 +26,78 @@
...
     {
         $builder
-            ->add('enable_order_return', SwitchType::class, [
+            ->add(static::FIELD_ENABLE_ORDER_RETURN, SwitchType::class, [
+                'label' => $this->trans(
+                    'Enable returns',
+                    'Admin.Orderscustomers.Feature'
+                ),
+                'help' => $this->trans(
+                    'Would you like to allow merchandise returns in your shop?',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php]: We could get rid of this tooltip. The label "Enable returns" speaks for itself. :-)

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php [hunk 2]
```diff
@@ -26,36 +26,78 @@
...
+                    'Admin.Orderscustomers.Help'
+                ),
                 'required' => false,
+                'multistore_configuration_key' => 'PS_ORDER_RETURN',
             ])
-            ->add('order_return_period_in_days', IntegerType::class, [
+            ->add(static::FIELD_ORDER_RETURN_PERIOD_IN_DAYS, IntegerType::class, [
+                'label' => $this->trans(
+                    'Time limit of validity',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php]:
```suggestion
                    'Return window',
```

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php [hunk 3]
```diff
@@ -26,36 +26,78 @@
...
+                    'Admin.Orderscustomers.Feature'
+                ),
+                'help' => $this->trans(
+                    'How many days after the delivery date does the customer have to return a product?',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php]:
```suggestion
                    'Maximum number of days to return a product after the delivery date.',
```

**Review by Julievrz — COMMENTED** [review]
A few wording suggestions

**Review by Julievrz — DISMISSED** [review]
Thanks, wording ✔️

**Review by atomiix — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Sell/CustomerService/MerchandiseReturn/Blocks/options.html.twig [hunk 4]
```diff
@@ -23,40 +23,24 @@
...
 
-{{ form_start(merchandiseReturnsOptionsForm, {action: path('admin_merchandise_returns_save_options')}) }}
+{{ form_start(merchandiseReturnsOptionsForm, {
+  attr : {class: 'form'},
+  action: path('admin_merchandise_returns_save_options')
+}) }}
   <div class="card">
     <h3 class="card-header">
-      {{ 'Merchandise return (RMA) options'|trans({}, 'Admin.Orderscustomers.Feature') }}
+      <i class="material-icons">settings</i> {{ 'Merchandise return (RMA) options'|trans({}, 'Admin.Orderscustomers.Feature') }}}}
```
Comment (saulaski) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/CustomerService/MerchandiseReturn/Blocks/options.html.twig]: Why 2 extra closing brackets ?
```suggestion
      <i class="material-icons">settings</i> {{ 'Merchandise return (RMA) options'|trans({}, 'Admin.Orderscustomers.Feature') }}
```

File: src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php [hunk 5]
```diff
@@ -26,36 +26,32 @@
...
-     */
-    public function __construct(ConfigurationInterface $configuration)
-    {
-        $this->configuration = $configuration;
-    }
+    private const CONFIGURATION_FIELDS = [
+        MerchandiseReturnOptionsType::FIELD_ENABLE_ORDER_RETURN,
+        MerchandiseReturnOptionsType::FIELD_ORDER_RETURN_PERIOD_IN_DAYS,
+        MerchandiseReturnOptionsType::FIELD_ORDER_RETURN_PREFIX,
+    ];
```
Comment (saulaski) [inline src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php]: Architecture bad practice: Core components should never use Bundle constants. The Bundle should depend on the Core, not the other way around.
Instead public constants should be defined here in Core, and then used by the form type

**Review by saulaski — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Sell/CustomerService/MerchandiseReturn/Blocks/options.html.twig (hunk 4 above)
Comment (Progi1984) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/CustomerService/MerchandiseReturn/Blocks/options.html.twig]: Fixed

**Review by saulaski — APPROVED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php [hunk 6]
```diff
@@ -26,36 +26,74 @@
...
+    public const FIELD_ORDER_RETURN_PREFIX = 'order_return_prefix';
+
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('enable_order_return', SwitchType::class, [
+            ->add(static::FIELD_ENABLE_ORDER_RETURN, SwitchType::class, [
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php:52]: It's not exactly conventional to use const for form fields But why not

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php [hunk 7]
```diff
@@ -26,36 +26,74 @@
...
+                'label' => $this->trans(
+                    'Enable returns',
+                    'Admin.Orderscustomers.Feature'
+                ),
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php:56]: I believe you forgot the help message here `Would you like to allow merchandise returns in your shop?` that was initially integrated in the form template

Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Sell/CustomerService/MerchandiseReturn/MerchandiseReturnOptionsType.php:56]: Ah no it was asked by @Julievrz my bad

**Review by jolelievre — APPROVED** [review]
Thanks @Progi1984

**Review by PierreRambaud — APPROVED** [review]

File: src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php [hunk 8]
```diff
@@ -26,36 +26,31 @@
...
-
-    /**
-     * @param ConfigurationInterface $configuration
-     */
-    public function __construct(ConfigurationInterface $configuration)
-    {
-        $this->configuration = $configuration;
-    }
+    private const CONFIGURATION_FIELDS = [
+        'enable_order_return',
```
Comment (tegbessou) [inline src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php:38]: Why don't you use const in MerchandiseReturnOptionsType, to define configuration fields?

File: src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php [hunk 9]
```diff
@@ -26,36 +26,31 @@
...
+        'order_return_period_in_days',
```
Comment (tegbessou) [inline src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php:39]: Why don't you use const in MerchandiseReturnOptionsType, to define configuration fields?

File: src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php [hunk 10]
```diff
@@ -26,36 +26,31 @@
...
+        'order_return_prefix',
```
Comment (tegbessou) [inline src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php:40]: Why don't you use const in MerchandiseReturnOptionsType, to define configuration fields?

Comment (saulaski) [reply src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php:40]: As I commented previously, it's related to architecture good practice.
Core layer is not supposed to depend on the Bundle. But the Bundle is allowed to use constants defined in Core.

Comment (tegbessou) [reply src/Core/MerchandiseReturn/Configuration/MerchandiseReturnOptionsConfiguration.php:40]: Sorry I don't see your remark

**Review by marwachelly — APPROVED** [review]
Hello,

PR is tested. The checkboxes are well displayed.
image
https://watch.screencastify.com/v/zxELZVMZw7l3c8TAtjPb
PR is QA approuved.
Thanks,

**Comment by atomiix** [comment]
Thanks @Progi1984 & @marwachelly

**Pull request merged**